web/__init__.py

```diff
@@ -98,6 +135,86 @@ def app_api_version():
     }
 
 
+def allowed_file(filename, ALLOWED_EXTENSIONS):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+def config_show():
+    config = app.config["config"]
+    with config.open() as cfg:
+        config_output = f'''<p>Account:</p><p>
+            user_id:    {cfg.account.user_id[:10]}...[REDACTED] <br/>
+            auth_token: {cfg.account.auth_token[:10]}...[REDACTED] <br/>
+            email:      {cfg.account.email} <br/>
+            region:     {cfg.account.region.upper()} </p
+        '''
+        config_output += '<p>Printers:</p><hr/>'
+        for p in cfg.printers:
+            config_output += f'''<p>
+                duid:      {p.p2p_duid} <br/>
+                sn:        {p.sn} <br/>
+                ip:        {p.ip_addr} <br/>
+                wifi_mac:  {cli.util.pretty_mac(p.wifi_mac)} <br/>
+                api_hosts: {', '.join(p.api_hosts)} <br/>
+                p2p_hosts: {', '.join(p.p2p_hosts)} <hr/></p>
+            '''
+        return config_output
+
+
+def config_import(loginFile):
+    file = open(loginFile, 'r')
+    config = app.config["config"]
+
+    # extract auth token
+    cache = libflagship.logincache.load(file.read())["data"]
+    auth_token = cache["auth_token"]
+
+    # extract account region
+    region = libflagship.logincache.guess_region(cache["ab_code"])
+
+    try:
+        newConfig = cli.config.load_config_from_api(auth_token, region, False)
+    except libflagship.httpapi.APIError as E:
+        flash(f"Config import failed: {E} (auth token might be expired: make sure Ankermake Slicer can connect, then try again", 'danger')
+        return redirect('/')
+    except Exception as E:
+        flash(f"Config import failed: {E}", 'danger')
+        return redirect('/')
+
+    try:
+        config.save("default", newConfig)
+    except Exception as E:
+        flash(f"Config import failed: {E}", 'danger')
+        return redirect('/')
+    flash("Configuration imported! Restarting...", 'success')
+    return redirect('/')
+
+
+@app.post('/api/config/upload')
+def app_api_config_upload():
+    ALLOWED_EXTENSIONS = set(['json'])
+    with tempfile.TemporaryDirectory(prefix='ankerctl_') as tmpdir:
+        config = app.config["config"]
+
+        if request.method == 'POST':
```

Comment:
We get here by hooking `@app.post`, so this is always true?

If we want to keep it (for clarity, or otherwise), negate the check and make it return.

```python
if request.method != "POST":
    return
```

(again, to avoid long indentation)

Reply:
I'm fine with removing the IF here if we are pretty sure that flask will always handle `app.post` as POST and will never accept a GET on the endpoint :)

Reply:
I'm fine with leaving it in, for the same reason - but let's make it early-return, instead of indenting everything :)